database/migration/src/main/resources/changesets/failure_metadata_scripts/guarantee_causedBy.sql

```diff
@@ -0,0 +1,13 @@
+INSERT INTO METADATA_ENTRY (WORKFLOW_EXECUTION_UUID, METADATA_KEY, CALL_FQN, JOB_SCATTER_INDEX, JOB_RETRY_ATTEMPT, METADATA_TIMESTAMP)
+SELECT WORKFLOW_EXECUTION_UUID, REPLACE(METADATA_KEY, "message", "causedBy[]"), CALL_FQN, JOB_SCATTER_INDEX, JOB_RETRY_ATTEMPT, METADATA_TIMESTAMP
+FROM METADATA_ENTRY AS t1
+WHERE METADATA_KEY LIKE '%failures[%]%message'
+AND NOT EXISTS (SELECT *
+             	FROM METADATA_ENTRY AS t2
+             	WHERE t2.WORKFLOW_EXECUTION_UUID = t1.WORKFLOW_EXECUTION_UUID
+             	  AND (t2.CALL_FQN = t1.CALL_FQN OR (t2.CALL_FQN IS NULL AND t1.CALL_FQN IS NULL))
+             	  AND (t2.JOB_SCATTER_INDEX = t1.JOB_SCATTER_INDEX OR (t2.JOB_SCATTER_INDEX IS NULL AND t1.JOB_SCATTER_INDEX IS NULL))
+             	  AND (t2.JOB_RETRY_ATTEMPT = t1.JOB_RETRY_ATTEMPT OR (t2.JOB_RETRY_ATTEMPT IS NULL AND t1.JOB_RETRY_ATTEMPT IS NULL))
+                  AND t2.METADATA_KEY LIKE REPLACE(t1.METADATA_KEY, "message", "causedBy%")
```

Comment:
ToL: Similar preference for `CONCAT` instead of `REPLACE`, and `:` prefixing for keys.